src/Microsoft.Health.Dicom.Core/Features/Common/ParallelEnumerable.cs

```diff
@@ -0,0 +1,127 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.Common;
+
+internal sealed class ParallelEnumerationOptions
+{
+    public int MaxBufferedItems { get; init; } = 100;
+
+    public int MaxDegreeOfParallelism { get; init; } = Environment.ProcessorCount * 4;
```

Comment:
I've updated the configs to be less complicated with their defaults, but the duplicates are there because we have a service-level configuration and a method options object that mirrors the existing `ParallelOptions` class.

Reply:
The default also doesn't need to be capped because `Parallel` will handle that for us based on the task scheduler